lib/board.rb

```diff
@@ -0,0 +1,110 @@
+# rubocop:disable Metrics/CyclomaticComplexity
+# rubocop:disable Metrics/MethodLength
+# rubocop:disable Metrics/PerceivedComplexity
+# rubocop:disable Naming/VariableNumber
+# require_relative './main.rb'
+
+class Board
+  attr_accessor :grid
+
+  def initialize
+    @grid = [
+      %w[1 2 3],
+      %w[4 5 6],
+      %w[7 8 9]
+    ]
+  end
+
+  def valid?(pos)
+    pos.each do |ele|
+      return false if (0..2).include?(ele) == false
+    end
+    true
+  end
+
+  def empty?(pos)
+    %w[1 2 3 4 5 6 7 8 9].include?(@grid[pos[0]][pos[1]])
+  end
+
+  def place_mark(position, mark)
+    if valid?(position) && empty?(position)
+      @grid[position[0]][position[1]] = mark
+    else
+      puts 'Invalid mark'
+      puts 'Try again'
```

Comment:
- [ ] It is a good practice to place all `puts`, `print` and `chomps` in the main. Kindly ensure to move the `puts` into the `main.rb`